src/Geometry/PUMLReader.h

```diff
@@ -94,6 +88,81 @@ class PUMLReader : public MeshReader
 	static int FIRST_FACE_VERTEX[4];
 };
 
+/*
+ * Possible types of boundary conditions for SeisSol.
+ */
+enum class BCType {
+	internal, external, unknown
+};
+
+/**
+ * Decodes the boundary condition tag into a BCType.
+ */
+inline BCType bcToType(int id) {
+	if (id == 0 || id == 3 || id > 64) {
+		return BCType::internal;
+	} else if (id == 1 || id == 2 || id == 5 || id == 6) {
+		return BCType::external;
+	} else {
+		return BCType::unknown;
+	}
+}
+
+/**
+ * Decodes the boundary condition tag into a string representation.
+ */
+inline std::string bcToString(int id) {
+	if (id == 0) { return std::string("regular"); }
+	else if (id == 1) { return std::string("free surface"); }
+	else if (id == 2) { return std::string("free surface with gravity"); }
+	else if (id == 3) { return std::string("dynamic rupture"); }
+	else if (id == 5) { return std::string("absorbing"); }
+	else if (id == 6) { return std::string("periodic"); }
+	else if (id > 64) {
+		std::stringstream s;
+		s << "fault-tagging (" << id << ")";
+		return s.str();
+	} else { return std::string(""); }
```

Comment:
This should include the id of the cell - right now you're missing e.g. bc 7 (analytical/dirichlet). I also have some branches with internal boundary conditions.